src/fields_global_stats.c

```diff
@@ -0,0 +1,92 @@
+#include "fields_global_stats.h"
+#include "config.h"
+
+void FieldsGlobalStats_UpdateStats(FieldSpec *fs, int toAdd) {
+  if (fs->types & INDEXFLD_T_FULLTEXT) {  // text field
+    RSGlobalConfig.fieldsStats.numTextFields += toAdd;
+  } else if (fs->types & INDEXFLD_T_NUMERIC) {  // numeric field
+    RSGlobalConfig.fieldsStats.numNumericFields += toAdd;
+  } else if (fs->types & INDEXFLD_T_GEO) {  // geo field
+    RSGlobalConfig.fieldsStats.numGeoFields += toAdd;
+  } else if (fs->types & INDEXFLD_T_VECTOR) {  // vector field
+    RSGlobalConfig.fieldsStats.numVectorFields += toAdd;
+    if (fs->vectorOpts.vecSimParams.algo == VecSimAlgo_BF)
+      RSGlobalConfig.fieldsStats.numVectorFieldsFlat += toAdd;
+    else if (fs->vectorOpts.vecSimParams.algo == VecSimAlgo_HNSWLIB)
+      RSGlobalConfig.fieldsStats.numVectorFieldsHSNW += toAdd;
+  } else if (fs->types & INDEXFLD_T_TAG) {  // tag field
+    RSGlobalConfig.fieldsStats.numTagFields += toAdd;
+    if (fs->tagOpts.tagFlags & TagField_CaseSensitive) {
+      RSGlobalConfig.fieldsStats.numTagFieldsCaseSensitive += toAdd;
+    }
+  }
+
+  if (fs->options & FieldSpec_Sortable) {
+    if (fs->types & INDEXFLD_T_FULLTEXT) RSGlobalConfig.fieldsStats.numTextFieldsSortable += toAdd;
+    else if (fs->types & INDEXFLD_T_NUMERIC) RSGlobalConfig.fieldsStats.numNumericFieldsSortable += toAdd;
+    else if (fs->types & INDEXFLD_T_GEO) RSGlobalConfig.fieldsStats.numGeoFieldsSortable += toAdd;
+    else if (fs->types & INDEXFLD_T_TAG) RSGlobalConfig.fieldsStats.numTagFieldsSortable += toAdd;
+  }
+  if (fs->options & FieldSpec_NotIndexable) {
+    if (fs->types & INDEXFLD_T_FULLTEXT) RSGlobalConfig.fieldsStats.numTextFieldsNoIndex += toAdd;
+    else if (fs->types & INDEXFLD_T_NUMERIC) RSGlobalConfig.fieldsStats.numNumericFieldsNoIndex += toAdd;
+    else if (fs->types & INDEXFLD_T_GEO) RSGlobalConfig.fieldsStats.numGeoFieldsNoIndex += toAdd;
+    else if (fs->types & INDEXFLD_T_TAG) RSGlobalConfig.fieldsStats.numTagFieldsNoIndex += toAdd;
+  }
+}
```

Comment:
Lets use a switch case.
I think we remove the NoIndex fields. @K-jo, WDYT?

Reply:
@ashtul Is it possible to do switch case to & operator?

Reply:
You are right.
We can have multi typed fields